### ISSUE TITLE
macOS 10.13.6: Symbol not found: _posix_spawn_file_actions_addchdir_np

### ISSUE THREAD

**Issue by jmoorman** [ziglang/zig#11562]
### Zig Version

0.10.0-dev.2024+d127c1d59

### Steps to Reproduce

System Version: macOS 10.13.6
Kernel Version: Darwin 17.7.0
```
zig init-exe
zig build run
```

### Expected Behavior

`info: All your codebase are belong to us.`

### Actual Behavior

```
dyld: lazy symbol binding failed: Symbol not found: _posix_spawn_file_actions_addchdir_np
  Referenced from: /Users/jm/testproj/zig-cache/o/69820f5080914c7866f7b19659e510bc/build
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _posix_spawn_file_actions_addchdir_np
  Referenced from: /Users/jm/testproj/zig-cache/o/69820f5080914c7866f7b19659e510bc/build
  Expected in: /usr/lib/libSystem.B.dylib
```

This issue does not reproduce with zig version 0.9.0-dev.866+6a6e2cd64

**Comment by coastwise** [comment]
I have the same version of macOS (10.13.6; the last one compatible with my hardware) and just encountered the same error after updating zig. :(

**Comment by pdckxd** [comment]
Also can reproduce from my side. While Rust still provides support for earily MacOS. Would be better for Zig to support them as well. I know some deverlopers have to use old machines with old OSes due to lower income.

**Comment by Mouvedia** [comment]
binary: bun 0.1.17
command: bun test
macOS: 10.13.6

**Comment by kkHAIKE** [comment]
`posix_spawn_file_actions_addchdir_np` available in macos 10.15, zig use it too spawn_and_chdir child process, i think it need change to libc solution before macos  10.15 (golang does this)

**Comment by kubkon** [comment]
I believe https://github.com/ziglang/zig/pull/17152 fixes this. I will thus close the issue but please do let me know if this is not the case and will reopen.
